types/mssql/index.d.ts

```diff
@@ -391,3 +391,8 @@ export declare class PreparedStatement extends events.EventEmitter {
 }
 
 export declare class PreparedStatementError extends MSSQLError {}
+
+export declare function connect(config: config, callback: (err?: any, pool?: ConnectionPool) => void): void;
+export declare function connect(config: config): Promise<ConnectionPool>;
+export declare function connect(connectionString: string, callback: (err?: any, pool?: ConnectionPool) => void): void;
+export declare function connect(connectionString: string): Promise<ConnectionPool>;
```

Comment:
1. While we're at it, can we add typings where `connect()` accepts 0 arguments if there's an existing global connection?

    **References:**
    - https://github.com/tediousjs/node-mssql/blob/89492de2227f0876100ecc7af2f5d352c348d4ab/lib/global-connection.js#L17
    - https://github.com/tediousjs/node-mssql/blame/89492de2227f0876100ecc7af2f5d352c348d4ab/README.md#L358-L361

2. Perhaps we can also consider merging and simplifying the type definitions? IMO this would improve readability,  and is what's followed by the existing types:

    <details>
    <summary>Existing examples</summary>

    https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f3130558c77ec1d970840c9451d49c20b20f81fa/types/mssql/index.d.ts#L259
    https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f3130558c77ec1d970840c9451d49c20b20f81fa/types/mssql/index.d.ts#L283
    https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f3130558c77ec1d970840c9451d49c20b20f81fa/types/mssql/index.d.ts#L322-L323
    https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f3130558c77ec1d970840c9451d49c20b20f81fa/types/mssql/index.d.ts#L362-L363
    https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f3130558c77ec1d970840c9451d49c20b20f81fa/types/mssql/index.d.ts#L381-L382

    </details>

```suggestion
export declare function connect(config?: config | string, callback: (err?: any, pool?: ConnectionPool) => void): void;
export declare function connect(config?: config | string): Promise<ConnectionPool>;
```

Reply:
I applied your first point.

On the second one, I copied the constructor signature for `Connection` class which is what this function hooks into:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bab884457e31721eb800af45ca9a013fb0054d63/types/mssql/index.d.ts#L222-L223

It would then probably make sense to consolidate those as well then?